src/backend/distributed/utils/maintenanced.c

```diff
@@ -262,6 +267,14 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		CHECK_FOR_INTERRUPTS();
 
 		/*
+		 * We clear the metadata cache before every iteration because otherwise
```

Comment:
Can we put an XXX here and open an issue? Because something here is very much not right with this.